javascripts/discourse/widgets/layouts-category-list.js

```diff
@@ -67,6 +67,8 @@ export default layouts.createLayoutsWidget('category-list', {
     const { childCategories } = this.attrs;
     if (!childCategories || !category || !childCategories[category.slug])
       return [];
+    if (settings.child_categories_default_state === 'hidden') return [];
```

Comment:
If we're never going to show the child categories with this config, we shouldn't load them at all in the ``layouts-category-list`` initializer. This code can then remain the same as ``childCategories`` would be empty.